### PR TITLE
[DRAFT][compiler] Support RmsNorm Op and Fuse

### PR DESCRIPTION
**Pull request by seockho-kim**
To support LLM,

- Added RmsNorm Op to circle format
- RmsNorm fuse , kernel implemented.

ONE-DCO-1.0-Signed-off-by: seockho.kim [seockho.kim@samsung.com](mailto:seockho.kim@samsung.com)

### CODE REVIEW

**Comment by seockho-kim** [comment]
For https://github.com/Samsung/ONE/issues/13964

File: compiler/common-artifacts/exclude.lst [hunk 1]
```diff
@@ -178,3 +178,4 @@ tcgenerate(CircleFullyConnected_U4_002)
 tcgenerate(GRU_000) # luci-interpreter does not support custom GRU
 tcgenerate(InstanceNorm_000)
 tcgenerate(InstanceNorm_001)
+tcgenerate(RmsNorm_000)
```
Comment (seanshpark) [inline compiler/common-artifacts/exclude.lst:181]: as this is exclude list, if `luci-interpreter` supports RmsNorm, plz remove this line.

**Comment by seanshpark** [comment]
Great work!

Overall codes looks OK. Reviewers will take a deeper reading with each PRs. Let's begin posting PRs :) 
- single module per PR
- single context of change per PR

would be nice.

File: compiler/common-artifacts/exclude.lst (hunk 1 above)
Comment (seockho-kim) [reply compiler/common-artifacts/exclude.lst:181]: Okay, there is no error after removing that line. :)

File: res/CircleSchema/0.8/circle_schema.fbs [hunk 2]
```diff
@@ -623,6 +624,7 @@ union BuiltinOptions {
   BCQGatherOptions = 252,
   BCQFullyConnectedOptions = 253,
   InstanceNormOptions = 254,
+  RmsNormOptions = 255,
```
Comment (glistening) [inline res/CircleSchema/0.8/circle_schema.fbs:627]: Please use `250`, instead of `255`.

File: compiler/one-cmds/onelib/constant.py [hunk 3]
```diff
@@ -48,6 +48,7 @@ class CONSTANT:
         'fuse_instnorm',
         'fuse_prelu',
         'fuse_gelu',
+        'fuse_rmsnorm',
```
Comment (jinevening) [inline compiler/one-cmds/onelib/constant.py:51]: Could you exclude this option in O1? Our backend compiler is not ready for this change.

Comment (seockho-kim) [reply compiler/one-cmds/onelib/constant.py:51]: Oh, I couldn't think of that. 
I'll update it :)

File: res/CircleSchema/0.8/circle_schema.fbs (hunk 2 above)
Comment (seockho-kim) [reply res/CircleSchema/0.8/circle_schema.fbs:627]: Okay, I'll update it. 
ref) RightShiftOptions = 126, so we have 100+ rooms.

**Comment by seockho-kim** [comment]
@glistening advised to remove activation function in RmsNorm parameters, so it will be modified.

**Comment by seanshpark** [comment]
If you want `void init(loco::Graph *g) override { NodeGraphletT<luci::CircleRmsNorm>::init(g); }` not to be in single line, add `//` like
```cxx
void init(loco::Graph *g) override
{
  //
  NodeGraphletT<luci::CircleRmsNorm>::init(g);
}
```

**Comment by seockho-kim** [comment]
Done

**Pull request closed without merging**